salt/_modules/metalk8s_volumes.py

```diff
@@ -377,6 +442,76 @@ def is_cleaned_up(self):
     def clean_up(self):
         return  # Nothing to do
 
+class RawBlockDeviceBlock(RawBlockDevice):
+    def __init__(self, volume):
+        super(RawBlockDeviceBlock, self).__init__(volume)
+        # Detect which kind of device we have: a real disk, only a partition or
+        # an LVM volume.
+        name = device_name(self.path)
+        self._partition = None
+        for symlink in glob.glob('/dev/disk/by-id/dm-uuid-LVM-*'):
+            realpath = os.path.realpath(symlink)
+            if os.path.basename(realpath) == name:
+                self._kind = DeviceType.LVM
+                return
```

Comment:
Minor, it looks like the code to find a LVM symlink here and in `persistent_path()` could be factorized in a function:
```
def get_lvm_path(self):
    name = device_name(self.path)
    for symlink in glob.glob('/dev/disk/by-id/dm-uuid-LVM-*'):
        realpath = os.path.realpath(symlink)
        if os.path.basename(realpath) == name:
            return symlink
    return None
```
```suggestion
        self._partition = None
        if self.get_lvm_path():
            self._kind = DeviceType.LVM
            return
```

Reply:
Good point, I've added a commit to factorize it!